iothub/device/src/Transport/ErrorDelegatingHandler.cs

```diff
@@ -219,7 +220,11 @@ async Task<T> ExecuteWithErrorHandlingAsync<T>(Func<Task<T>> asyncOperation, boo
                             throw new IotHubClientTransientException("Transient error occurred, please retry.", ex);
                         }
 
-                        await this.Reset(openCompletionBeforeOperationStarted, handlerBeforeOperationStarted).ConfigureAwait(false);
+                        if (reset)
```

Comment:
/cc @tameraw @abhipsaMisra 

We need to investigate all other cases for this as it appears to be an old design-level bug: InternalClient (Device/Module) has a circular dependency through some of the callbacks. 
Instead of having the state machine spread out between Error, Retry and InternalClient callbacks we should push the logic into the correct handler.